cloudprovider/aws/aws_provider.go

```diff
@@ -0,0 +1,70 @@
+package aws
+
+import (
+	"encoding/json"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/rancher/agent/utilities/config"
+	"os"
+	"path"
+	"time"
+)
+
+const (
+	regionLabel           = "ec2.region"
+	availabilityZoneLabel = "ec2.availability_zone"
+	location              = "aws"
+	infoFile              = "info.json"
+	tempFile              = "temp.json"
+)
+
+type Info struct {
+	Label map[string]string `json:"label,omitempty"`
+}
+
+func CallProvider() {
```

Comment:
CallProvider is a bad function name. It doesn't do a good job describing what you're doing. Maybe GetCloudProviderInfo or something like that.